### PR TITLE
Refactoring

### PR DESCRIPTION
**Pull request by Borjianamin98**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,8 +2,49 @@
 
 A **simple to use** graphical LL(1) parser generator
 
+
+## Getting Started
+
+For run this project on your own local machine or server you should install java and maven.
```
Comment (IYP-Programer-Yeah) [inline README.md]:
```suggestion
This project uses Java, hence to run it, you machine should be Java enabled. To compile this project you are going to need maven, however, if you don't want to compile from source, there are releases you can use.
```

File: README.md [hunk 2]
```diff
@@ -2,8 +2,49 @@
...
+You can download jar which is released and use it without compiling it with maven
+
+### Prerequisites
+
+For installing dependencies for this project do this instructions.
```
Comment (IYP-Programer-Yeah) [inline README.md]:
```suggestion
Following prerequisites are required, depending on your usage.
```

File: README.md [hunk 3]
```diff
@@ -2,8 +2,49 @@
...
+
+Install java
```
Comment (IYP-Programer-Yeah) [inline README.md]:
```suggestion
#### Java

To execute this project you are going to need Java JDK, which you can install via the following command, if you are on a Linux machine.

```

File: README.md [hunk 4]
```diff
@@ -2,8 +2,49 @@
...
+```
+sudo apt install openjdk-8-jdk
+```
+
+Install maven
```
Comment (IYP-Programer-Yeah) [inline README.md]:
```suggestion
#### Maven

You are going to need Maven only if you are compiling from source. You can install Maven using the following command if you are on a Linux machine.

```

File: README.md [hunk 5]
```diff
@@ -2,8 +2,49 @@
...
+```
+sudo apt install maven
+```
+### Installing
```
Comment (IYP-Programer-Yeah) [inline README.md]:
```suggestion

### Compilation

You can compile the code via the following command.
```

File: README.md [hunk 6]
```diff
@@ -2,8 +2,49 @@
...
+
+You can compile PGen with below command:
```
Comment (IYP-Programer-Yeah) [inline README.md]:
```suggestion
```

File: README.md [hunk 7]
```diff
@@ -2,8 +2,49 @@
...
+```
+mvn package
+```
+After that, You will see this: `target/PGen-*.*-jar-with-dependencies.jar`. Run that with bellow command
```
Comment (IYP-Programer-Yeah) [inline README.md]:
```suggestion

The result of compilation is a `.jar` file, that you can find in the `target` director. The file is going to be named `PGen-*.*-jar-with-dependencies.jar` where the `*.*` is going to be the version of the binary.
```

Comment (IYP-Programer-Yeah) [reply README.md]: `where the `*.*` is going to be the version of the binary.` Is this part correct about the naming? if not modify accordingly.

File: README.md [hunk 8]
```diff
@@ -2,8 +2,49 @@
...
+```
```
Comment (IYP-Programer-Yeah) [inline README.md:43]:
```suggestion

### Execution

You can run the executable with the following command.

```
```

**Comment by Borjianamin98** [comment]
I updated README as you said.

**Comment by Borjianamin98** [comment]
Update README with your offeres

File: README.md (hunk 6 above)
Comment (IYP-Programer-Yeah) [reply README.md]: This need to be applied, the line above this addresses this.

**Review by IYP-Programer-Yeah — APPROVED** [review]

**Comment by Borjianamin98** [comment]
I didn’t understand. I have to change something??

**Comment by IYP-Programer-Yeah** [comment]
nah, thing are good i am merging

**Pull request merged**